driver-core/src/main/java/com/datastax/driver/core/BoundStatement.java

```diff
@@ -270,7 +270,7 @@ public BoundStatement setRoutingKey(ByteBuffer routingKey) {
      */
     @Override
     public String getKeyspace() {
-        return statement.getPreparedId().metadata.size() == 0 ? null : statement.getPreparedId().metadata.getKeyspace(0);
+        return statement.getPreparedId().getVariables().size() == 0 ? null : statement.getPreparedId().getVariables().getKeyspace(0);
```

Comment:
With the refactoring I came across this line and thought: this would return `null` for statements without bound variables. But we could also inspect the resultset metadata (i.e. the returned columns) to find out the keyspace.